examples/twittermap/web/public/javascripts/searchbar/controllers.js

```diff
@@ -2,9 +2,43 @@ angular.module('cloudberry.util', ['cloudberry.common'])
   .controller('SearchCtrl', function($scope, $window, cloudberry, cloudberryConfig) {
     $scope.search = function() {
       if ($scope.keyword && $scope.keyword.trim().length > 0) {
-        cloudberry.parameters.keywords = $scope.keyword.trim().split(/\s+/);
-        // skip the empty query for now.
-        cloudberry.query(cloudberry.parameters);
+
+          //stopword filtering feature
+          String.prototype.removeStopWords = function() {
+              var x;
+              var word;
+              var cleansed_string = this.valueOf();
+              var Hstop_words = external_stop_words();
+
+              //Split out all the individual words in the phrase
+              var words = cleansed_string.split(" ");
+
+              //Review each token
+              for(x=0; x<words.length; x++){
+                  word = words[x];
+
+                  //If matches, remove it from the keywords
+                  if(Hstop_words.has(word.toLowerCase()))
+                  {
+                      // Remove the word from the keywords
+                      cleansed_string = cleansed_string.replace(word, "");
```

Comment:
Rather than removing stopwords from the given keywords, it's better to create the final keyword.

if (!Hstop_words.has(word.toLowerCase())) {
   newString = newString + word;
}